Shared/Realm.Sync.Shared/Credentials.cs

```diff
@@ -0,0 +1,162 @@
+
+            internal const string Password = "password";
+
+            internal const string Identity = "identity";
+
+            internal const string IsAdmin = "isAdmin";
+        }
+
+        public static Credentials Custom(string identityProvider, string userIdentifier,
+            IReadOnlyDictionary<string, object> userInfo)
+        {
+            return new Credentials
+            {
+                IdentityProvider = identityProvider,
+                Token = userIdentifier,
+                UserInfo = userInfo
+            };
+        }
+
+        public static Credentials Debug()
+        {
+            return new Credentials
+            {
+                IdentityProvider = Providers.Debug
+            };
+        }
+
+        public static Credentials Facebook(string facebookToken)
+        {
+            if (facebookToken == null)
+            {
+                throw new ArgumentNullException(nameof(facebookToken));
+            }
+
+            return new Credentials { IdentityProvider = Providers.Facebook, Token = facebookToken };
+        }
+
+        public static Credentials Google(string googleToken)
+        {
+            if (googleToken == null)
+            {
+                throw new ArgumentNullException(nameof(googleToken));
+            }
+
+            return new Credentials { IdentityProvider = Providers.Google, Token = googleToken };
+        }
+
+        public static Credentials Twitter(string twitterToken)
+        {
+            if (twitterToken == null)
+            {
+                throw new ArgumentNullException(nameof(twitterToken));
+            }
+
+            return new Credentials { IdentityProvider = Providers.Twitter, Token = twitterToken };
+        }
+
+        public static Credentials UsernamePassword(string username, string password, bool createUser)
+        {
+            return new Credentials 
+            { 
+                IdentityProvider = Providers.Password,
+                Token = username,
+                UserInfo = new Dictionary<string, object> { [Keys.CreateUser] = createUser, [Keys.Password] = password }
+            };
+        }
+
+        public static Credentials AccessToken(string accessToken, string identity, bool isAdmin = false)
+        {
+            return new Credentials
+            {
+                IdentityProvider = Providers.AccessToken,
+                Token = accessToken,
+                UserInfo = new Dictionary<string, object> { [Keys.Identity] = identity, [Keys.IsAdmin] = isAdmin }
+            };
+        }
+
+        public string IdentityProvider { get; private set; }
+
+        public string Token { get; private set; }
+
+        public IReadOnlyDictionary<string, object> UserInfo { get; private set; } = new Dictionary<string, object>();
+
+        private Credentials()
+        {
+        }
+
+        internal JsonObject ToJson()
+        {
+            var user_info = new JsonObject();
```

Comment:
It's not, but it's the name for the field in the JSON dictionary in the authentication protocol.